### DISCUSSION TITLE
Repository Split

### DISCUSSION THREAD

**Discussion by plebhash** [stratum-mining/stratum#1071] [General]
During the call today (`July 23 2024`) there was rough consensus on the need for eventually splitting SRI into multiple repositories.

This discussion should be used for proposals and discussions on this topic.

**Comment by pavlenex** [comment]
Thanks for logging this one!

**Comment by plebhash** [comment]
here's a proposal I've been working on for a while:

![SRI stack](https://github.com/user-attachments/assets/e88a4f57-0f44-4753-abe1-66cead51ce58)

**Reply by plebhash** [reply]
I've been preparing this for a long time, but I feel now was the right moment to share

I will take some time later to write here a more lengthy description of all the components.

**Reply by plebhash** [reply]
PR https://github.com/stratum-mining/stratum/pull/1066/ (Roles integration tests) is relevant for this proposal, so I'd like to ask @jbesraa to provide some potential adjustments on the diagram above so we make sure our visions are aligned

**Comment by bit-aloo** [comment]
A few design patterns that can be considered during refactoring: https://refactoring.guru/design-patterns/rust

**Comment by plebhash** [comment]
closing in favor of https://github.com/stratum-mining/stratum/discussions/1069#discussioncomment-10290201
